APCAppCore/APCAppCore/Library/Categories/NewDataCollector/Data Sink/APCPassiveDataSink.m

```diff
@@ -0,0 +1,472 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+#import "APCPassiveDataSink.h"
+#import "APCAppCore.h"
+#import "zipzap.h"
+#import "APCDataVerificationClient.h"
+#import "CMMotionActivity+Helper.h"
+
+static NSString *const  kCollectorFolder    = @"newCollector";
+static NSString *const  kUploadFolder       = @"upload";
+static NSString *const  kIdentifierKey      = @"identifier";
+static NSString *const  kStartDateKey       = @"startDate";
+static NSString *const  kEndDateKey         = @"endDate";
+static NSString *const  kInfoFilename       = @"info.json";
+static NSString *const  kCSVFilename        = @"data.csv";
+static NSString *const  kItemIdentifier     = @"displacement";
+static long long        kKBPerMB            = 1024;
+static long long        kBytesPerKB         = 1024;
+static NSUInteger       kSecsPerMin         = 60;
+static NSUInteger       kMinsPerHour        = 60;
+static NSUInteger       kHoursPerDay        = 24;
+static NSUInteger       kDaysPerWeek        = 7;
+
+@implementation APCPassiveDataSink
+
+/*********************************************************************************/
+#pragma mark - Abstract methods from delegate
+/*********************************************************************************/
+- (void)didRecieveUpdatedValuesFromCollector:(NSArray*)quantitySamples
+{
+    __weak typeof(self) weakSelf = self;
+    
+    [quantitySamples enumerateObjectsUsingBlock: ^(id quantitySample, NSUInteger __unused idx, BOOL * __unused stop)
+    {
+        __typeof(self) strongSelf = weakSelf;
+        
+        [strongSelf processUpdatesFromCollector:quantitySample];
+    }];
+}
+
+- (void)didRecieveUpdatedValueFromCollector:(id)result
+{
+    [self processUpdatesFromCollector:result];
+}
+
+- (void)didRecieveUpdateWithLocationManager:(CLLocationManager*) __unused manager withUpdateLocations:(NSArray*) __unused locations
+{
+    [self checkIfCSVStructureHasChanged];
+}
+
+/*********************************************************************************/
+#pragma mark - Abstract methods
+/*********************************************************************************/
+
+- (void)processUpdatesFromCollector:(id)quantitySample
+{
+    __weak typeof(self) weakSelf = self;
+    
+    [self.healthKitCollectorQueue addOperationWithBlock:^{
```

Comment:
The parameter is poorly named: potentially an array of samples may be processed by this method.